tensortrade/environments/trading_environment.py

```diff
@@ -156,254 +185,89 @@ def reward_scheme(self, reward_scheme: Union[RewardScheme, str]):
-            reward (float): An size corresponding to the benefit earned by the action taken this timestep.
-            done (bool): If `True`, the environments is complete and should be restarted.
-            info (dict): Any auxiliary, diagnostic, or debugging information to output.
-        """
-        order = self._take_action(action)
-
-        observation = self._next_observation()
-        reward = self._get_reward()
-        done = self._done()
-        info = self._info(order)
-
         if self._enable_logger:
-            self.logger.debug('Order: {}'.format(order))
-            self.logger.debug('Observation: {}'.format(observation))
-            self.logger.debug('P/L: {}'.format(self._portfolio.profit_loss))
+            self.logger.debug('Order:       {}'.format(order))
+            self.logger.debug('Observation: {}'.format(obs))
+            self.logger.debug('P/L:         {}'.format(self._portfolio.profit_loss))
             self.logger.debug('Reward ({}): {}'.format(self.clock.step, reward))
             self.logger.debug('Performance: {}'.format(self._portfolio.performance.tail(1)))
 
         self.clock.increment()
 
-        return observation, reward, done, info
+        return obs, reward, done, info
 
-    def reset(self) -> pd.DataFrame:
+    def reset(self) -> np.array:
         """Resets the state of the environments and returns an initial observation.
 
         Returns:
             The episode's initial observation.
         """
         self.clock.reset()
-
-        if not self._exchange.is_live:
-            if self._initial_balances is not None:
-                self._portfolio._wallets = {}
-
-                for balance in self._initial_balances:
-                    self._portfolio.add((self._exchange, balance.instrument, balance.size))
-            else:
-                self._initial_balances = self._portfolio.total_balances
-
-        self._action_scheme.reset()
-        self._reward_scheme.reset()
-        self._exchange.reset()
-        self._portfolio.reset()
-        self._broker.reset()
-
-        observation = self._next_observation()
+        self.feed.reset()
+        self.action_scheme.reset()
+        self.reward_scheme.reset()
+        self.portfolio.reset()
+        self.broker.reset()
+        self.history.reset()
+
+        # Get next observation
+        row = self.feed.next()
+        if not self.use_internal:
+            row = {k: row[k] for k in self._external_keys}
+        self.history.push(row)
+        obs = self.history.observe()
 
         self.clock.increment()
 
-        return observation
+        return obs
 
     def render(self, mode='none'):
         """Renders the environment via matplotlib."""
         if mode == 'log':
             self.logger.info('Performance: ' + str(self._portfolio.performance))
         elif mode == 'chart':
-            if self.viewer is None and hasattr(self.exchange, '_pre_transformed_data'):
```

Comment:
Why was this removed? The replacement code is not correct.

If the idea is to not require Matplotlib, then within `MatplotlibTradingChart`, we can import `matplotlib` within the `__init__` statement.